source/val/validate_annotation.cpp

```diff
@@ -260,19 +260,15 @@ spv_result_t ValidateDecorationTarget(ValidationState_t& _, SpvDecoration dec,
       }
       break;
     case SpvDecorationBuiltIn:
-      if (target->opcode() != SpvOpVariable &&
-          !spvOpcodeIsConstant(target->opcode())) {
-        return _.diag(SPV_ERROR_INVALID_DATA, inst)
-               << "BuiltIns can only target variables, structure members or "
-                  "constants";
-      }
-      if (_.HasCapability(SpvCapabilityShader) &&
-          inst->GetOperandAs<SpvBuiltIn>(2) == SpvBuiltInWorkgroupSize) {
+      if (target->opcode() != SpvOpVariable) {
```

Comment:
I think this check could be greatly improved or just moved into the `validate_builtins.cpp` as well but for sake of not overloading this PR too much will leave for a post-PR fix